src/util/types.cpp

```diff
@@ -100,8 +100,9 @@ isString32Valid(std::string const& str)
     return true;
 }
 
+template <typename T>
 bool
-isAssetValid(Asset const& cur)
+isAssetValid(T const& cur)
```

Comment:
I'm not sure if this is necessary. This method is only defined in the cpp file, so it should be difficult to use incorrectly. What do you think?

Reply:
Not sure I agree. A future maintainer of this file could reasonably use `isAssetValid<ChangeTrustAsset>` instead of `isChangeTrustAssetValid` and it wouldn't be easy to notice this during code review.

Reply:
That's a valid point. I updated this so `isAssetValid` is the only method that needs to be used. The fixup commit is large though because I added the ledger version as a parameter.